version: "3.9"

services:
  auth_service:
    image: auth_service
    build:
      context: ./
      dockerfile: dockerfile.dev
    ports:
      - "4000:4000"
    volumes:
      - "./src:/usr/src/app/src"
      - "./.env:/usr/src/app/.env"
    restart: on-failure

  mongodb:
      image: mongo:4.4.6-bionic
      command: mongod --auth
      env_file:
        - .env
      environment:
        - MONGO_INITDB_ROOT_USERNAME=${MONGO_INITDB_ROOT_USERNAME}
        - MONGO_INITDB_ROOT_PASSWORD=${MONGO_INITDB_ROOT_PASSWORD}
        - MONGO_INITDB_DATABASE=${MONGO_INITDB_DATABASE}
        - MONGO_INITDB_USERNAME=${MONGO_INITDB_USERNAME}
        - MONGO_INITDB_PASSWORD=${MONGO_INITDB_PASSWORD}
        - APP_INIT_USER=${APP_INIT_USER}
        - APP_INIT_PW=${APP_INIT_PW}
        - PUID=1000
        - PGID=1000
      volumes:
        - ./.docker/mongodb/mongo-init.sh:/docker-entrypoint-initdb.d/mongo-init.sh:ro
        - ./devdb:/data/db
      ports:
        - 27017:27017
      restart: on-failure
      logging:
        driver: none